### PR TITLE
Move to pymavlink in place of dronekit-python

### PR DESCRIPTION
**Pull request by peterbarker**
... also use `sys.stderr` for progress messages

### CODE REVIEW

**Comment by peterbarker** [comment]
Chief advantage of this PR is that you don't need to wait for dronekit-python to be "ready" - which is mostly parameter-downloads, which this script doesn't need.  The vehicle abstraction which dronekit-python provides isn't necessary or useful in this script - and the extra verbiage required to send messages is pretty silly.

Second major advantage here is the stderr change - it means when started from systemd the logs get the error messages in good order.

File: scripts/d4xx_to_mavlink.py [hunk 1]
```diff
@@ -185,51 +187,76 @@
...
+    progress("INFO: Debugging option enabled")
 else:
-    print("INFO: Debugging option DISABLED")
+    progress("INFO: Debugging option DISABLED")
 
 ######################################################
 ##  Functions - MAVLink                             ##
 ######################################################
 
+def MAVLinkLoop(conn, callbacks):
```
Comment (rmackay9) [inline scripts/d4xx_to_mavlink.py:232]: Looks like we're switching to camel case when the rest of the script uses underlines?

File: scripts/d4xx_to_mavlink.py [hunk 2]
```diff
@@ -600,14 +582,30 @@ def get_local_ip():
...
+
 ######################################################
 ##  Main code starts here                           ##
 ######################################################
 
-print("INFO: Connecting to vehicle.")
-while (not vehicle_connect()):
-    pass
-print("INFO: Vehicle connected.")
+progress("INFO: pyrealsense2 version: %s" % str(rs.__version__))
```
Comment (rmackay9) [inline scripts/d4xx_to_mavlink.py:592]: It's greating having the pyrealsense version displayed here so this is a good addition I think.  I'm less sure about the change from "Connecting to Vehicle" to "Starting MAVLink connection" and "STarting MAVLink thread".  Remember this is going to be seen by regular people and I'm not sure many of them know what MAVLink is and probably most don't know what a "thread" is.  Let's keep it simple.

File: scripts/d4xx_to_mavlink.py [hunk 3]
```diff
@@ -629,28 +627,34 @@ def get_local_ip():
...
+    glib_loop = GLib.MainLoop()
+    glib_thread = threading.Thread(target=glib_loop.run, args=())
+    glib_thread.start()
 else:
     send_msg_to_gcs('RTSP not streaming')
 
 sched.start()
 
+main_loop_should_quit = False
+def sigint_handler(sig, frame):
```
Comment (rmackay9) [inline scripts/d4xx_to_mavlink.py:648]: this is nice, it means that someone could "Ctrl-C" the script and make it stop I think.  Are there are interrupt handlers that we need to be careful of?  having the camera stop working is not great in some situations.

Comment (peterbarker) [reply scripts/d4xx_to_mavlink.py:669]: Only the default interrupt handler is present before adding this patch.

Yes, this is supposed to clean the camera connection a bit and perhaps make it lock up less...

File: scripts/d4xx_to_mavlink.py (hunk 2 above)
Comment (peterbarker) [reply scripts/d4xx_to_mavlink.py]: I've moved this back to "INFO: Starting Vehicle communications"

**Comment by thien94** [comment]
Thank you very much for the insightful comments and suggestions. I guess the dronekit-python related parts are not necessary for now.

I will try it out on my system before merging.

**Comment by rishabsingh3003** [comment]
@thien94 while you are on this, there is another small issue that I have found... it would be great if you could verify it as well.
When you enable debugging options in your script and diplay the output image... I belive the green line is at the wrong location?
https://youtu.be/VjnAIbJb-Ks - Here is video of this problem, you'd see no data is detected when my hand is over the green line and its actually much below the line where the data is being sent.
I have it fixed here: https://github.com/peterbarker/vision_to_mavros/pull/1/commits/3f0e4799a0176b6ad29e791feed01489312af3e6

But it would be great if you could confirm this once.. it's mostly a resizing issue. Thanks!

**Comment by thien94** [comment]
@rishabsingh3003 I can't seem to replicate the problem with the current code on master. What settings did you use for the experiment? Did the issue happen before or after the changes in this PR?

**Comment by rishabsingh3003** [comment]
@thien94 this experiment is on your master, with no changes at all except - I limited the Range to [0.1,0.5] so that it only sees my hand skips the rest of the depth data, which can be seen in the video. 
I am also using d435i 
Everything else is on default. I.e I simply run - `python3 d4xx_to_mavlink.py --debug_enable 1`

**Comment by thien94** [comment]
Thank you @peterbarker. It works well on my system, merging now.

@rmackay9 I will try to apply the useful changes in this PR into the t265_to_mavlink.py script as well.

@rishabsingh3003 Thank you for the info. I will add your fix in the next commit.

**Comment by rmackay9** [comment]
@thien94, great, thanks very much!   really happy the changes will go into the t265 script as well ('cuz this is actually what I use most)!

**Pull request merged**